### DISCUSSION TITLE
Linguflex from openai to local

### DISCUSSION THREAD

**Discussion by clustercd** [KoljaB/Linguflex#5] [Ideas]
Tested [LocalAIVoiceChat](https://github.com/KoljaB/LocalAIVoiceChat) which is brillant and really fluent, impressive job and well documented.
Started with Linguflex and asking here if the same local model as LocalAIVoiceChat can be add to replace the Openai key ?
In fact the best world ever should be a mixt between the both Linguflex and LocalAIVoiceChat, local one miss module to interact with world and Linguflex miss privacy with openai.
brillant job anyway and very well documentation

**Comment by KoljaB** [comment]
Linguflex 2 can run local. Open the [settings](https://github.com/KoljaB/Linguflex/blob/lingu-2.0-preview/lingu/settings.yaml) file and under local_llm section set local_llm parameter to "true". Specify name of local model in the same section under model_name parameter, download the model and set the model_path parameter to the download path.

**Reply by clustercd** [reply]
Thx for answer. Indeed on my way to test it, will need time to cover all the functionalities this time, it's huge works ! a bit screwed on some point for now, where to declare the cudnn, it's a new env variable ? And when selected Coqui for voice in UI raised    super().__init__(open(name, mode)) FileNotFoundError: [Errno 2] No such file or directory: 'models/xtts\\speakers_xtts.pth',
Last blocking point it's regarding the language itself, i imagine realtimetts working only in english so impossible to switch in fr : RealTimeSTT: faster_whisper - WARNING - The current model is English-only but the language parameter is set to 'fr'; using 'en' instead.I had the same in [LocalAIVoiceChat](https://github.com/KoljaB/LocalAIVoiceChat) impossible to modify the voice recorder to fr instead of en, so had to ask in en and receive impressive response in perfect french. I will continue to try to reach as it's pretty sure i missed smth.
thx

**Reply by KoljaB** [reply]
*where to declare the cudnn, it's a new env variable ?*
Only install cuDNN from [here](https://developer.nvidia.com/rdp/cudnn-archive), no env variable required for this (if I remember correctly).

*No such file or directory: 'models/xtts\speakers_xtts.pth'*
Either run download_models.py or if you already have XTTS installed somewhere open [settings](https://github.com/KoljaB/Linguflex/blob/lingu-2.0-preview/lingu/settings.yaml) file and change speech/xtts_model_path parameter to point to the folder containing your xtts model.

On your last points, both RealtimeTTS and RealtimeSTT are heavily multilingual. For RealtimeSTT you need to take a multilingual whisper model. So don't take like "tiny.en" or so because these are english. Take "medium" or "large-v2". RealtimeSTT supports [100 languages](https://github.com/openai/whisper/blob/main/whisper/tokenizer.py#L11-L110), specify your language shortcut under language parameter. 

RealtimeTTS language support depends on the engine. With CoquiEngine for example it can do 17 languages: English (en), Spanish (es), French (fr), German (de), Italian (it), Portuguese (pt), Polish (pl), Turkish (tr), Russian (ru), Dutch (nl), Czech (cs), Arabic (ar), Chinese (zh-cn), Japanese (ja), Hungarian (hu), Korean (ko) and Hindi (hi). Same as with RealtimeSTT you have to set the language shortcut as a parameter in the engine, otherwise it will speak with english accent.

**Reply by clustercd** [reply]
thx for explanation and indeed works as a charm now for the realtimeTTS.
Only point still blocking for installation of Coqui is a \TTS\utils\generic_utils.py", line 142, in get_user_data_dir
    dir_, _ = winreg.QueryValueEx(key, "Local AppData")
FileNotFoundError: [WinError 2] impossible to find the location. I tried with absolute path and r without success.

**Reply by KoljaB** [reply]
For example:
If the model.pth file of the main XTTS 2.0.2 model is in D:\Projekte\TestLingu\Linguflex\models\xtts\v2.0.2 then you would enter either D:\Projekte\TestLingu\Linguflex\models\xtts (absolute) or models/xtts (relative) into xtts_model_path parameter of settings.yaml.

If you submit a folder and there is not v2.0.2 subfolder in it, RealtimeTTS should create that v2.0.2 folder and start downloading the model for its own. 

If all that does not work, I need more info. Like was there already a model.pth downloaded? The full logging stacktrace would be helpful. Because I'm not sure why this:
\TTS\utils\generic_utils.py", line 142, in get_user_data_dir
dir_, _ = winreg.QueryValueEx(key, "Local AppData")
is logged. That's just the tts library asking the registry for the users local app folder, hard to say what exactly is going wrong there and what it has to do with the FileNotFoundError without more logging.

**Reply by clustercd** [reply]
Found the issue. no value in user local-> replaced with local machine key and it's working well now.
